pyrightconfig.json

```diff
@@ -6,6 +6,7 @@
   "enableTypeIgnoreComments": true,
   // Exclude from scanning when running pyright
   "exclude": [
+    ".git/",
```

Comment:
What's this for? Can we ignore `.*? Or can we not have pyright or something, how could it possibly get confused in this way?

Reply:
Git stores branch names as files on disc
Some branch names end in `.py`
pyright tries to scan them as python files and fails

This only affects local development because the CI doesn't fetch all branches. It's also already on main so it'll disappear from this PR after I resolve the conflict.

Doesn't affect mypy because it always ignores .git.
It would affect Ruff if the config used `exclude` instead of `extend-exclude`